### PR TITLE
Validate etcd backup and restore credentials

### PR DESCRIPTION
**Pull request by moadqassem**
**What does this PR do / Why do we need it**:
During the creation of the a seed object, there is no check for the secret which has the credentials for the etcd-backup-and-restore controller. Which means, if the controller was enabled, with any credentials, it will try to create backups regardless whether the s3 bucket credentials existence. This would lead to creating `etcdbackupconfig` that fails all the time and also prevent the cluster of being deleted, since the controller fails to clean the etcdbackupconfig objects. Thus the cluster will be in a stuck-in-deletion state. 

This PR makes sure that, if the seed enables the EtcdBackupAndRestore controller, it validates that the secret exists and if not, it will reject it. 
**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by xrstf** [comment]
Validating the credentials is good, but what happens if the credentials simply change out of Kubermatic's control? Like someone rotating the AWS S3 credentials without updating the KKP configuration? Should our cleanup code, which would currently be blocked by this, be more forviging?

**Comment by moadqassem** [comment]
> Validating the credentials is good, but what happens if the credentials simply change out of Kubermatic's control? Like someone rotating the AWS S3 credentials without updating the KKP configuration? Should our cleanup code, which would currently be blocked by this, be more forviging?

You will be blocked anyway by the failing etcd backup

File: pkg/webhook/seed/validation.go [hunk 1]
```diff
@@ -190,6 +193,12 @@ func (v *validator) validate(ctx context.Context, obj runtime.Object, isDelete b
 		if _, exists := subject.Spec.EtcdBackupRestore.Destinations[subject.Spec.EtcdBackupRestore.DefaultDestination]; !exists {
 			return fmt.Errorf("invalid etcd backup configuration: default destination %q does not exist", subject.Spec.EtcdBackupRestore.DefaultDestination)
 		}
+
+		etcdBackupSecret := corev1.Secret{}
+		if err := seedClient.Get(ctx, types.NamespacedName{Name: resources.EtcdRestoreS3CredentialsSecret,
```
Comment (lsviben) [inline pkg/webhook/seed/validation.go]: we need to check if the credentials of each destination exist, not `resources.EtcdRestoreS3CredentialsSecret`.

`resources.EtcdRestoreS3CredentialsSecret` is a leftover of old code, ill remove it in a separate PR.

Comment (moadqassem) [reply pkg/webhook/seed/validation.go]: Done

**Comment by moadqassem** [comment]
> I just realized one thing. We should only check if Credentails exist, IF they are set.
> 
> Because a destination can exist without credentials, although it wont work, and backup/restore jobs wont get started. Which is better then bad credentials, where jobs are started, but stuck because the secret is missing.

Hmm not exactly, so what will happen is, a default etcdbackupconfig is gonna be created, regardless of the secret being there or not. That backup will live in the cluster namespace forever with a status Failed. And since the credentials are wrong, then the controller won't be able to cleanup those which eventually would lead to cluster being stuck in termination forever.

File: pkg/webhook/seed/validation.go [hunk 2]
```diff
@@ -190,6 +192,16 @@ func (v *validator) validate(ctx context.Context, obj runtime.Object, isDelete b
 		if _, exists := subject.Spec.EtcdBackupRestore.Destinations[subject.Spec.EtcdBackupRestore.DefaultDestination]; !exists {
 			return fmt.Errorf("invalid etcd backup configuration: default destination %q does not exist", subject.Spec.EtcdBackupRestore.DefaultDestination)
 		}
+
+		for _, dest := range subject.Spec.EtcdBackupRestore.Destinations {
+			if dest.Credentials != nil {
+				etcdBackupSecret := corev1.Secret{}
+				if err := seedClient.Get(ctx, types.NamespacedName{Name: dest.Credentials.Name,
+					Namespace: dest.Credentials.Namespace}, &etcdBackupSecret); err != nil {
+					return fmt.Errorf("invalid etcd backup configuration: invalid etcd backup configs for destination %v: %w", dest.Credentials.Name, err)
```
Comment (xrstf) [inline pkg/webhook/seed/validation.go]: > invalid etcd backup configuration: invalid etcd backup configs for destination %v

sounds a bit stuttery. WDYT about

> invalid etcd backup configuration: invalid destination %s: %w

and replace %s with the name of the destination, not the same of the credential Secret.

Comment (moadqassem) [reply pkg/webhook/seed/validation.go]: There is no name for a destination

Comment (moadqassem) [reply pkg/webhook/seed/validation.go]: That's why I used the credential ;). Either cases I updated the error message

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**